digital_asset_types/src/rpc/asset.rs

```diff
@@ -459,3 +462,42 @@ pub struct TokenAccount {
     pub close_authority: Option<String>,
     pub extensions: Option<Value>,
 }
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Default, JsonSchema)]
+#[serde(default)]
+pub struct UiTokenAmount {
+    pub amount: String,
+    pub decimals: u8,
+    #[serde(rename = "uiAmount")]
+    pub ui_amount: Option<f64>,
+    #[serde(rename = "uiAmountString")]
+    pub ui_amount_string: String,
+}
+
+impl RpcTokenAccountBalance {
+    pub fn from(token_account: token_accounts::Model, mint_decimals: u8) -> Self {
+        let ui_amount = if mint_decimals > 0 {
+            token_account.amount as f64 / 10u64.pow(mint_decimals as u32) as f64
+        } else {
+            token_account.amount as f64
+        };
+        let amount = UiTokenAmount {
+            amount: token_account.amount.to_string(),
+            decimals: mint_decimals,
+            ui_amount: Some(ui_amount),
+            ui_amount_string: ui_amount.to_string(),
+        };
+        Self {
+            address: bs58::encode(token_account.pubkey).into_string(),
+            amount,
+        }
+    }
+}
```

Comment:
Sorry I didn't see the dependency on mint_decimals. I am fine with having it inlined like before sorry about that.